### DISCUSSION TITLE
Openssl 3.0 provider support

### DISCUSSION THREAD

**Discussion by gabmaralnxp** [Azure/azure-iot-sdk-c#2561] [General]
Hello,

I'm trying to find some examples for how to use openssl 3.0 provider (https://www.openssl.org/docs/man3.0/man7/provider.html) and I cannot find anything relevant.
All I can see is that openssl engine is supported, but that is legacy from openssl 1.1.1 and deprecated in openssl 3.0,
At the same time, openssl 1.1.1 is EOL and not recommended to be used anymore.

Is there an example similar to prov_dev_client_ll_x509_sample.c, where SAMPLE_OPENSSL_ENGINE can be enabled, but which makes use of openssl provider instead?

Thank you!

**Comment by ericwolz** [comment]
Openssl 3.0 is already supported which uses the deprecated interfaces.

#2371
